pkg/apis/kubeapiserver/v1alpha1/types.go

```diff
@@ -0,0 +1,53 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	operatorsv1alpha1api "github.com/openshift/api/operator/v1alpha1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// KubeApiserverConfig provides information to configure kube-apiserver
+type KubeApiserverConfig struct {
+	metav1.TypeMeta `json:",inline"`
+}
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ServiceCertSignerOperatorConfig provides information to configure an operator to manage the service cert signing controllers
+type KubeApiserverOperatorConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata" protobuf:"bytes,1,opt,name=metadata"`
+
+	Spec   KubeApiserverOperatorConfigSpec   `json:"spec" protobuf:"bytes,2,opt,name=spec"`
+	Status KubeApiserverOperatorConfigStatus `json:"status" protobuf:"bytes,3,opt,name=status"`
+}
+
+type KubeApiserverOperatorConfigSpec struct {
+	operatorsv1alpha1api.OperatorSpec `json:",inline" protobuf:"bytes,1,opt,name=operatorSpec"`
+
+	// kubeApiserverConfig holds a sparse config that the user wants for this component.  It only needs to be the overrides from the defaults
+	// it will end up overlaying in the following order:
+	// 1. hardcoded default
+	// 2. this config
+	KubeApiserverConfig runtime.RawExtension `json:"kubeApiserverConfig" protobuf:"bytes,2,opt,name=kubeApiserverConfig"`
```

Comment:
@deads2k I guess this will also need to allow specify admin-defined volumes?

Reply:
Compare above: `KubeApiserverConfig`. It will be a sparse config, to be be edited by the admin.

Possibly, we put the volumes into a separate fields here though if `KubeApiserverConfig` resembles our classic OpenShift config.